### PR TITLE
lab16

### PR DESCRIPTION
**Pull request by stephenml101**

### CODE REVIEW

**Comment by stephenml101** [comment]
- [ x] Top-level README “Table of Contents” is updated
 - [x ] README for this challenge is complete
       - [ x] Summary, Description, Approach & Efficiency, Solution
       - [ x] Picture of whiteboard
       - [ x] Link to code
 - [ x] Feature tasks for this challenge are completed
 - [ x] Unit tests written and passing
       - [x ] “Happy Path” - Expected outcome
       - [ x] Expected failure
       - [ x] Edge Case (if applicable/obvious)

File: java/datastructures/lib/src/test/java/datastructures/tree/BinaryTreeTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    tree.add(2);
+    tree.add(6);
+    tree.add(5);
+    tree.add(11);
+    tree.add(5);
+    tree.add(9);
+    tree.add(4);
+
+    int expectedMax = 11;
+    assertEquals(Optional.of(expectedMax), tree.getMax());
```
Comment (lucy-gelderloos) [inline java/datastructures/lib/src/test/java/datastructures/tree/BinaryTreeTest.java:63]: Not sure what Optional.of is doing here - once I removed it, the test passed correctly.
